### ISSUE TITLE
Some Questions

### ISSUE THREAD

**Issue by wly-ysu** [zhyever/Monocular-Depth-Estimation-Toolbox#29]
Hello, thank you very much for your excellent work. I have some problems with the data set. When I download and copy the data, some pictures are damaged.error is NoneType. I wrote a detection code, but I encountered a problem. I hope to get your guidance.

code：
import os
import cv2.cv2

f = open('./data/kitti/benchmark_train.txt', 'r', encoding='UTF-8')
# print(f.read())
data_list = f.read().split('\n')
for i, data in enumerate(data_list):
    if i % 1 == 0:
        print(i)
    if len(data) == 0:
        continue
    data_info=data.split(' ')
    img = cv2.cv2.imread(os.path.join('./data/kitti/gt_depth', data_info[0]))
    print(data_info[0])
    h = img.shape[0]
    img = cv2.cv2.imread(os.path.join('./data/kitti/gt_depth', data_info[0][0 : 11], data_info[1]))
    print(data_info[1])
    h = img.shape[0]

f.close()

**Comment by HelloWorld4096** [comment]
I get an error when training the model, I can't find the picture, but I can find it in the path, I add print(filename) on line 226 of your Monocular-Depth-Estimation-Toolbox/depth/datasets/pipelines/loading.py 
![2022-06-25 14-01-07 的屏幕截图](https://user-images.githubusercontent.com/94941040/175760893-0b2ae873-18bc-4221-b16a-d0356df705c1.png)
![2022-06-25 14-02-08 的屏幕截图](https://user-images.githubusercontent.com/94941040/175760898-5cfbd8ea-e661-477b-837c-297cbafcd6fe.png)
![2022-06-25 14-03-40 的屏幕截图](https://user-images.githubusercontent.com/94941040/175760902-7f75f3d2-dcaa-44e5-9ac2-fa4066660698.png)
,

**Comment by zhyever** [comment]
I wonder if this error occurs generally? or is just caused by the specific missed ones?

**Comment by HelloWorld4096** [comment]
I had a similar problem when I trained the kitti dataset before. At that time, I also wrote a test file for reading the path. After replacing the file, there is generally no problem, but this time I can't judge whether it is a special case, because if my test does not Problem, the file does exist. This is contradictory, so I thought maybe there was something wrong with my test program. please take a look

**Comment by zhyever** [comment]
That's weird. How about utilizing the absolute path? Replace the `data/kitti` with your absolute path? Moreover, try `ls` to check if the file exits in that folder?

**Comment by HelloWorld4096** [comment]
thank you for your reply。I am now checking whether the files under the path are damaged, and I have changed the absolute path recommended by you. If there are still problems later, I will continue to ask you。

**Comment by zhyever** [comment]
That's OK. Good luck.

**Comment by HelloWorld4096** [comment]
Hello, I have solved this problem, I think it is an example, nonetype caused by the damage of the picture, using the above monitoring code (configure an opencv environment), after I replace the picture, I can train the code (bts-kitti_24e, time ratio Depth completion is much faster, thank you for your guidance), I have a short time to learn monocular depth estimation, I am confused about the naming method of some files, and I don't quite understand what it means, such as r50 (ResNet-50), *_24e , *_kitti_benchmark_*, *_22k_*w7_* I don’t know what it means, and there is another confusion. I want to know how to run the graphics card to full capacity. I checked the memory of my graphics card and found that the memory is only half used. I set it to be yes sample_per_gpu=2, workers_per_gpu= 4,8 (no change after modification), hope to get your guidance
![2022-06-26 07-16-21 的屏幕截图](https://user-images.githubusercontent.com/94941040/175793773-66c8c613-c7de-4181-afae-4f7ac436a9bd.png)
![2022-06-26 07-17-26 的屏幕截图](https://user-images.githubusercontent.com/94941040/175793775-3514bd8c-7c20-4f61-b099-a468ee88fd16.png)
.

**Comment by zhyever** [comment]
24e: 24 epochs. kitti_benchmark: utilizing kitti_benchmark split and preparing for submission to the official online benchmark. See [Tips.5](https://github.com/zhyever/Monocular-Depth-Estimation-Toolbox/blob/main/docs/inference.md#test-a-dataset). 22k: ImageNet22k pre-train. w7: window_size=7 for swin transformer.

`sample_per_gpu` and `workers_per_gpu` corresponds to `batch_size` and `num_workers` in dataloader, respectively. So if you wanna make full use of your card, increase the `sample_per_gpu` instead of `workers_per_gpu`. At the same time, the learning rate may need to be changed to adapt to the batch size changing.

**Comment by wly-ysu** [comment]
Thanks for your reply, it's clear at once.

**Comment by wly-ysu** [comment]
Hello, the device I am using is 2*2080ti. I trained your model, and I see that your training is all using v100 * 8. I have set sample_per_gpu to 1, but the error still shows that the GPU memory is insufficient. How do you use the inference time of a single graphics card test, I want to learn to modify the configuration and let it run on my computer.Looking forward to your reply.

**Comment by zhyever** [comment]
I need more information to help you. What configs? Which model?

**Comment by wly-ysu** [comment]
I'll elaborate on the problem I'm having, and I'm going to use the following line of command.
bash ./tools/dist_train.sh configs/depthformer/depthformer_swinl_22k_w7_kitti_benchmark.py 2 --work-dir work_dirs/depthformer/depthformer_swinl_22k_w7_kitti_benchmark.py
To train your model, I adjusted sample_per_gpu = 1 in the morning, but the GPU memory is still not enough.

**Comment by wly-ysu** [comment]
Hello, I try to use swint, this one works, another comment out /config/datasets/kitti.py line 6 crop_size=( (352, 704)) or line 14 (dict(type='RandomCrop', crop_size=(352, 704))), can't run, all show that the video card memory is insufficient, it should be that the 2*2080ti GPU video memory is too small to run. I have a new problem, when I run with this line of code (bash ./tools/dist_train.sh configs/dpt/dpt_vit-b16_kitti.py 2 --work-dir work_dirs/dpt/dpt_vit-b16_kitti.py), Report image error, jx_vit_base_p16_224-80ecf9dd.pth I have used this line of code (python tools/model_converters/vit2depth.py https://github.com/rwightman/pytorch-image-models/releases/download/v0.1- vitjx/jx_vit_base_p16_224-80ecf9dd.pth pretrain/jx_vit_base_p16_224-80ecf9dd.pth) for conversion, and also placed in the .cache/torch/hub/checkpoints directory. Hope to get your reply.
![2022-06-29 16-19-17 的屏幕截图](https://user-images.githubusercontent.com/94941040/176390366-24a07cc2-478b-41b5-88f2-dc65314d2bc1.png)

**Comment by zhyever** [comment]
Check the instruction for DPT. You need to run a script to convert pre-trained parameter dict.

**Comment by wly-ysu** [comment]
![QQ截图20220702082135](https://user-images.githubusercontent.com/94941040/176979646-896b60e8-1084-40b2-baf0-ab1c3eca0560.png)Thank you for your guidance, I have completed the conversion according to the instructions on the picture, but there is still a problem, I will try to convert again immediately.

**Comment by wly-ysu** [comment]
Hello, I would like to know if Adabins uses the pre-training set. I have downloaded the pre-training set in the picture in your project. I did not find the code related to the path of the data set. Please guide me and look forward to your reply 
![QQ截图20220706152612](https://user-images.githubusercontent.com/94941040/177493817-5b1a8445-cc33-458c-8ce1-86bb2b8e0b61.png)
.

**Comment by zhyever** [comment]
Please refer to [code](https://github.com/zhyever/Monocular-Depth-Estimation-Toolbox/blob/6f18fb8f1a198e8fd80ec87e88ab2da5d376f4ad/depth/models/backbones/efficientnet.py#L20), the encoder loads the pre-trained parameters as default.

**Comment by wly-ysu** [comment]
Thanks for the guide, I'll try again.

**Comment by wly-ysu** [comment]
I solved this problem by modifying config to --config in the parameters of tools/train.py

**Comment by duzongdong** [comment]
> ![QQ截图20220702082135](https://user-images.githubusercontent.com/94941040/176979646-896b60e8-1084-40b2-baf0-ab1c3eca0560.png)感谢您的指导，我已经按照图片上的说明完成了转换，但仍然存在问题，我会立即尝试再次转换。

**Comment by duzongdong** [comment]
Hello, I encountered the same problem when I was doing this step, how do you solve it

**Comment by duzongdong** [comment]
> 您好，我想知道 Adabins 是否使用预训练集。我已经下载了您项目中图片中的预训练集。我没有找到与数据集路径相关的代码。请指导我，期待您的回复 ![QQ截图20220706152612](https://user-images.githubusercontent.com/94941040/177493817-5b1a8445-cc33-458c-8ce1-86bb2b8e0b61.png) 。
Hello, I encountered the same problem when I was doing this step, how do you solve it？Thank you！

**Comment by duzongdong** [comment]
> ![QQ截图20220702082135](https://user-images.githubusercontent.com/94941040/176979646-896b60e8-1084-40b2-baf0-ab1c3eca0560.png)感谢您的指导，我已经按照图片上的说明完成了转换，但仍然存在问题，我会立即尝试再次转换。

Hello, I encountered the same problem when I was doing this step, how do you solve it?Thank you and look forward to your reply.
![image](https://github.com/zhyever/Monocular-Depth-Estimation-Toolbox/assets/112672589/647e8421-6805-4f68-82a8-401c79497edf)
